ui/dashboards/src/utils/variables.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useTemplateVariableValues, parseTemplateVariables, replaceTemplateVariables } from '@perses-dev/plugin-system';
+
+// Convenience hook for replacing template variables in a string
+export function useReplaceVariablesInString(str: string | null): string {
```

Comment:
Out of curiosity, why does this take a `null` value? Guessing there's an edge case it's helpful for.

Reply:
See comment above above about this and maybe switching it to `undefined`. Related, would it make sense for this function to return `undefined` if the value is `undefined` (instead of returning an empty string)?

Reply:
Yeah I went back and forth on this. I'll change it to undefined